internal/services/oracle/oracle_exascale_database_virtual_machine_cluster_resource.go

```diff
@@ -0,0 +1,622 @@
+					state.VirtualMachineFileSystemStorage = FlattenVMFileSystemStorage(props.VMFileSystemStorage)
+					state.VnetId = props.VnetId
+					state.Location = model.Location
+					state.Tags = pointer.From(model.Tags)
+					state.Zones = pointer.From(model.Zones)
+
+					state.BackupSubnetCidr = pointer.From(props.BackupSubnetCidr)
+					state.ClusterName = pointer.From(props.ClusterName)
+					state.DataCollectionOption = flattenExadbDataCollectionOptionInterface(metadata.ResourceData.Get("data_collection_option").([]interface{}))
+					state.Domain = pointer.From(props.Domain)
+					state.LicenseModel = string(pointer.From(props.LicenseModel))
+					state.NetworkSecurityGroupCidrs = FlattenNetworkSecurityGroupCidr(props.NsgCidrs)
+					state.Ocid = pointer.From(props.Ocid)
+					state.PrivateZoneOcid = pointer.From(props.PrivateZoneOcid)
+					state.SingleClientAccessNameListenerPortTcp = pointer.From(props.ScanListenerPortTcp)
+					state.SingleClientAccessNameListenerPortTcpSsl = pointer.From(props.ScanListenerPortTcpSsl)
+					state.SystemVersion = metadata.ResourceData.Get("system_version").(string)
+					state.TimeZone = pointer.From(props.TimeZone)
+					state.ZoneOcid = pointer.From(props.ZoneOcid)
+				}
+			}
+
+			return metadata.Encode(&state)
+		},
+	}
+}
+
+func (ExascaleDatabaseVirtualMachineClusterResource) Delete() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 30 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			client := metadata.Client.Oracle.OracleClient.ExadbVMClusters
+
+			id, err := exadbvmclusters.ParseExadbVMClusterID(metadata.ResourceData.Id())
+			if err != nil {
+				return err
+			}
+
+			if err = client.DeleteThenPoll(ctx, *id); err != nil {
+				return fmt.Errorf("deleting %s: %+v", *id, err)
+			}
+
+			return nil
+		},
+	}
+}
+
+func (ExascaleDatabaseVirtualMachineClusterResource) IDValidationFunc() pluginsdk.SchemaValidateFunc {
+	return exadbvmclusters.ValidateExadbVMClusterID
+}
+
+func flattenExadbDataCollectionOptionInterface(input []interface{}) []ExascaleDatabaseDataCollectionOptionModel {
+	output := make([]ExascaleDatabaseDataCollectionOptionModel, 0)
+	if len(input) == 0 || input[0] == nil {
+		return output
+	}
+	if m, ok := input[0].(map[string]interface{}); ok {
+		dataCollection := ExascaleDatabaseDataCollectionOptionModel{
+			IsDiagnosticsEventsEnabled: m["diagnostics_events_enabled"].(bool),
+			IsHealthMonitoringEnabled:  m["health_monitoring_enabled"].(bool),
+			IsIncidentLogsEnabled:      m["incident_logs_enabled"].(bool),
+		}
+		output = append(output, dataCollection)
+	}
+	return output
+}
+
+func expandNsgCidrs(input []NetworkSecurityGroupCidrModel) []exadbvmclusters.NsgCidr {
+	output := make([]exadbvmclusters.NsgCidr, 0, len(input))
+
+	for _, nsgCidr := range input {
+		portRangeValue := exadbvmclusters.PortRange{
+			Max: nsgCidr.DestinationPortRange[0].Max,
+			Min: nsgCidr.DestinationPortRange[0].Min,
```

Comment:
Can we validate the length first?

Reply:
You are right, then please add a comment above to mention this is guaranteed by the schema definition. Thx!